[CIVP-12081] BUG Allow non-seekable buffer inputs to file_to_civis

jkulzick opened the pull request:
… for a seekable file object

Included an additional check beyond HAS_TOOLBELT in civis.io.file_to_civis to ensure the file object is streamable (buf.seekable()). This should enable zip file objects to be uploaded using file_to_civis even when the requests_toolbelt is installed.

stephen-hoover commented on civis/tests/test_io.py:
```diff
@@ -99,6 +100,27 @@ def test_get_url_from_file_id(self, *mocks):
+    def test_file_to_civis(self, *mocks):
+        with tempfile.NamedTemporaryFile() as tmp:
+            tmp.write(b'a,b,c\n1,2,3')
+            tmp.flush()
```
This would also need a `tmp.seek(0)` so that we read the entire buffer, yes?

stephen-hoover commented on CHANGELOG.md:
```diff
@@ -15,6 +15,10 @@ This project adheres to [Semantic Versioning](http://semver.org/).
 
+## 1.5.3 - 2017-05-18
+### Changed
+- Changed criteria in ``civis.io.file_to_civis`` for streaming a file to include a check on if the file object is seekable.
```
The thing that users care about is that they can now provide file-like objects which aren't seekable, right? We used to error on that. I'd rephrase this entry to make that clear.

jkulzick commented:
Latest commit 6d792be:

1. Moved entry in CHANGELOG to the [unreleased] section and re-worded it
2. Updated comment block in civis.io.file_to_civis to also comment on new seekable criteria
3. removed change to version in _version
4. added tmp.seek(0) to test_file_to_civis prior to calling civis.io_file_to_civis

beckermr commented:
We need to make JIRA tickets for this stuff in the future, @jkulzick

jkulzick commented:
@beckermr @stephen-hoover 

Thanks for the jira examples @beckermr! I filed this jira. Hopefully it looks good
https://civisanalytics.atlassian.net/browse/CIVP-12081

beckermr commented:
LGTM on the JIRA! Thanks!
